js/source/vector_tile_source.js

```diff
@@ -71,9 +72,15 @@ VectorTileSource.prototype = util.inherit(Evented, {
             overscaling: overscaling,
             angle: this.map.transform.angle,
             pitch: this.map.transform.pitch,
-            collisionDebug: this.map.collisionDebug
+            collisionDebug: this.map.collisionDebug,
+            parentId: tile.parentId
         };
 
+        // request the tile parentID if it exists
+        if (tile.parentId) {
```

Comment:
afaict this is the first change I've seen that modifies existing behavior rather than extending it with a new source type.